### PR TITLE
Add migrator for data generation subclasses

### PR DESCRIPTION
**Pull request by brynnz22**
This PR addresses issue https://github.com/microbiomedata/berkeley-schema-fy24/pull/3 for making the subclasses of data generation. I went ahead and included `project.Makefile` changes so @turbomam can test easily, but when we are ready to merge, I can remove these changes and revert the project.makefile back to its original state.

This PR relies on, and needs to come after (and in the following order - I updated the `project.Makefile` so):

- `migrator_from_X_to_PR4` - transforms `omics_type` to match the new `AnalyteCategoryEnum`

- `migrator_from_X_to_PR2_and_PR24` - changes the class names (e.g. `omics_processing_set` to `data_generation_set`
 
- `migrator_from_X_to_PR10` - adds the `type` slot to all documents (it generically adds `type:DataGeneration` and does not take into account the subtypes (e.g. `type:NucleotideSequencing` and `type:MassSpectrometry`), I did not include this migrator in the `project.Makefile` because its not needed for testing, but does need to occur before this PR migration when we actually do the migration)

### CODE REVIEW

**Comment by turbomam** [comment]
I think I'd like to be more defensive here, too. This seems to change `part_of`s in any instance of any class. But couldn't we have instances of some class in which `part_of` shouldn't be changed? Unless a check like that is being applied somewhere else, then I'd like a check like that to be added here.

Then we can merge.

**Comment by brynnz22** [comment]
@turbomam I am confused. This PR does not do anything with `part_of`. Also, I wanted to test this once again once the other collection name PR gets merged so I can run that migration before this one to test.

**Comment by brynnz22** [comment]
@turbomam and @eecavanna this migrator is finally ready for review. Thanks!!

File: nmdc_schema/migrators/migrator_from_X_to_PR3.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        {'id': 123, 'analyte_category': 'nom', 'type': 'nmdc:MassSpectrometry'}
+        >>> m.specify_data_gen_type({'id': 234, 'analyte_category': 'metagenome', 'type': 'nmdc:DataGeneration'})
+        {'id': 234, 'analyte_category': 'metagenome', 'type': 'nmdc:NucleotideSequencing'}
+        """
+
+
+        nucleotide_seqs = ["metagenome", "metatranscriptome"]
+        mass_spec = ['metaproteome', 'metabolome', 'lipidome', 'nom']
+
+        if data_gen_doc["analyte_category"] in nucleotide_seqs:
```
Comment (eecavanna) [inline nmdc_schema/migrators/migrator_from_X_to_PR3.py:36]: If schema version `X` happens to not (I don't know whether it does) require that every document in the `data_generation_set` collection have a field named `analyte_category`, this line might produce a `KeyError` error due to the `analyte_category` key not existing in the `data_gen_doc` dictionary. That could be avoided by checking whether `"analyte_category"` is `in` `data_gen_doc` before trying to access `data_gen_doc["analyte_category"]`.

On the other hand, if schema version `X` does require that every document in the `data_generation_set` collection have a field named `analyte_category`, then this line is good to go, as far as I'm concerned.

**Review by eecavanna — APPROVED** [review]
Looks good to me! The only comment I left... may be a "non-issue" (I have lost track of what schema version `X` is, in this case).

File: nmdc_schema/migrators/migrator_from_X_to_PR3.py (hunk 1 above)
Comment (eecavanna) [reply nmdc_schema/migrators/migrator_from_X_to_PR3.py:36]: OK, thanks for confirming. I will resolve this conversation as the existing code is sufficient, in my opinion. The prerequisite for using this migrator is that the database conform to schema `X`. If schema `X` requires that all documents have that field, and a document happens to not have that field, the prerequisite was not met—which I consider to be out of the scope of this migrator (i.e. the job of ensuring the database meets the prerequisite).

**Pull request merged**